Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbDataStore.cs

```diff
@@ -54,7 +54,7 @@ public sealed class XldbDataStore : IDisposable
         /// Gets all the events of a certain type from the DB
         /// </summary>
         /// <returns>List of event objects recovered from DB </returns>
-        public IEnumerable<string> GetEventsByType(ExecutionEventId eventTypeID)
+        private IEnumerable<string> GetEventsByType(ExecutionEventId eventTypeID)

```

Comment:
I was debating back and forth whether to make this public or private since with the help of the public methods below, a user should never have to call this "GetEventsByType" method (thus private is better), but if they wanted to then the method should be public. What are your thoughts?

Reply:
Private is fine to reduce support surface, if users have a valid use case we can always make it public in the future.

Taking it away from public is always harder since it is a breaking change.

---
In reply to: [309472596](https://github.com/microsoft/BuildXL/pull/685#discussion_r309472596) [](ancestors = 309472596)

Reply:
Sure sounds good, I'll keep it private.